azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -338,18 +338,25 @@ public boolean shouldProxy() {
   }
 
   private void registerCustomCredential(final Props props, final Credentials hadoopCred, final
-  String userToProxy) {
+  String userToProxy, final Logger jobLogger) {
     String credentialClassName = "unknown class";
       try {
         credentialClassName = props
             .getString(Constants.ConfigurationKeys.CUSTOM_CREDENTIAL_NAME);
         logger.info("custom credential class name: " + credentialClassName);
-        final Class metricsClass = Class.forName(credentialClassName);
-
-        final Constructor[] constructors = metricsClass.getConstructors();
-        final CredentialProvider customCredential = (CredentialProvider) constructors[0]
-            .newInstance(hadoopCred, props);
-        customCredential.register(userToProxy);
+        final Class credentialClass = Class.forName(credentialClassName);
+
+        final Constructor[] constructors = credentialClass.getConstructors();
+        for (final Constructor constructor : constructors) {
+          if (constructor.getParameterCount() == 3) {
+            final CredentialProvider customCredential = (CredentialProvider) constructors[1]
```

Comment:
Should it be (CredentialProvider) constructor instead of (CredentialProvider) constructors[1]? Otherwise there is no point in comparing the constructors in the loop.

Reply:
Good catch! I removed the for loop and only look for the constructor accepting three parameters.